buildroot/board/meraki/ms220/overlay/bin/configd

```diff
@@ -0,0 +1,191 @@
+                false)
+                    echo "disabling port $port"
+                    echo "PORT $port, MODE off" >$CSPT/set_port_phy_cfgs
+                    ;;
+                *)
+                    echo "error: invalid port enabled value for port $port: $value"
+                    ;;
+                esac
+                ;;
+            lacp)
+                # TODO
+                # echo "AGGR 0, MEMBERS '9,10'" >$CSPT/add_link_aggr
+                echo "change port $port stp to $value (not implemented)"
+                ;;
+            stp)
+                # TODO: probably something with $CSPT/set_ports_stp_state
+                # echo -e "PORT $MAC/$i, ENABLED true, AUTOEDGE true, EDGE false, AUTOPTP true, PTP false, PRI 128, COST 0" > /click/stp/set_many_port_cfgs
+                echo "change port $port stp to $value (not implemented)"
+                ;;
+            link)
+                case $subfield in
+                speed)
+                    case $value in
+                    0)
+                        echo "removed forced speed for port $port"
+                        # TODO: should use port enabled value
+                        echo "PORT $port, MODE aneg" >$CSPT/set_port_phy_cfgs
+                        ;;
+                    100 | 1000 | 10000)
+                        echo "forcing port $port speed at $value"
+                        echo "PORT $port, MODE forced, FORCE_SPEED ${value}fdx" >$CSPT/set_port_phy_cfgs
+                        ;;
+                    *)
+                        echo "error: invalid forced port speed for port $port: $value"
+                        ;;
+                    esac
+                    ;;
+                established)
+                    # NO-OP: user should not be able to modify these fields
+                    ;;
+                *)
+                    echo "error: unknown link setting for port $port: $subfield"
+                    ;;
+                esac
+                ;;
+            vlan)
+                case $subfield in
+                pvid)
+                    set_vlan $port PVID $value
+                    ;;
+                allowed)
+                    set_vlan $port ALLOWED_VLANS $value
+                    ;;
+                tag_in)
+                    set_vlan $port ALLOW_TAGGED_IN $value
+                    ;;
+                untag_in)
+                    set_vlan $port ALLOW_UNTAGGED_IN $value
+                    ;;
+                *)
+                    echo "error: unknown vlan setting: $subfield for port $port"
+                    ;;
+                esac
+                ;;
+            poe)
+                case $subfield in
+                power)
+                    # NO-OP: user should not be able to modify these fields
+                    ;;
+                enabled)
+                    case $value in
+                    true)
+                        if [ $(pd690xx -l | grep "^ *$port\s" | grep "Disabled") ]; then
```

Comment:
On my MS220-8P with Build 20221115, when changing state of PoE (through Freeraki) to the non-set value, this will throw:
```sh
sh: Disabled: unknown operand
```
and nothing will happen. This is because the command outputs `1 Disabled at Critical 0.0` and causes an issue in the if block which then does not execute the `pd690xx` command.

In my case I changed this to:
```sh
$(pd690xx -l | grep "^ *$port\s" | grep "Disabled" > /dev/null)
                        if [ $? == 0 ]; then
```
I discard the output and if a grep result was found it will return exit code 0. This seems to work in my tests. Same concept applies to line 152.

Reply:
You can probably just change this statement to:
```
$(pd690xx -l | grep "^ *$port\s" | grep -c "Disabled")
```

`grep -c` will return the count with an exit status of 0 (count > 0) or 1 (count == 0)

Reply:
I updated `pd690xx` to only toggle the port if the state would change, the new firmware release is [20221127](https://watchmysys.com/files/meraki/ms220/postmerkOS-20221127.bin)

Since your statement calls `pd690xx` anyway, you can eliminate the `if` statement entirely and just run:
```
pd690xx -e $port
```

The call should be faster (since you have only one fork to `pd690xx` instead of also invoking `grep`) and will be a NOP unless the port state changed.

Reply:
Thanks for the feedback! In my testing (using this as the backend to a GUI) sh was way too slow. Unless anyone objects, I say we close this PR and I'll dust off my C impl and open a new one.

Reply:
If you're going to do this in C, link against `libpd690xx` for the PoE stuff. That was the motivation to rewrite it as a library.

Reply:
Yep, saw that -- really nice :+1: [most of the boilerplate is already there](https://github.com/halmartin/meraki-builder/compare/master...hall:meraki-builder:configd?expand=1#diff-bc6b04e0e77e2c5861b54d0f7b9ea574b9a13be2185ef891c4310f64233a02b1), just need to go through and clean up things and then test functionality.